### PR TITLE
feat(hardware): refactor tool_sensors to simplify and remove csvs

### PR DESCRIPTION
**Pull request by ryanthecoder**
# Overview
This PR removes much of the complexity in tool_sensors around the liquid probe routine.
Instead of having many different options of setting the bindings and building messages, this consolidates all of the previous options by only commanding the firmware one way, forwarding the sensor information to a new log file, and providing a way for the system to optionally grab the raw sensor data if they wish which is required for the hardware-testing scripts and possibly for future features.

We no longer write CSV files to the robot through this system and have removed all of the coded in paths for them. It also removes the "OutputOptions" data type that previously controlled the behavior variants.

Before each  variant had it's own quirks to behavior so now machine actions will always be the same regardless of how the tool_sensors method is called.

This also adds a new can message that sends sensor data in batches instead of 1 by 1 so we don't choke up the can bus as much.

### CODE REVIEW

File: api/src/opentrons/config/__init__.py [hunk 1]
```diff
@@ -202,6 +202,15 @@ class ConfigElement(NamedTuple):
         " absolute path, it will be used directly. If it is a "
         "relative path it will be relative to log_dir",
     ),
+    ConfigElement(
+        "sensor_log_file",
+        "SENSOR Log File",
```
Comment (sfoster1) [inline api/src/opentrons/config/__init__.py]:
```suggestion
        "Sensor Log File",
```

File: api/src/opentrons/hardware_control/backends/flex_protocol.py [hunk 2]
```diff
@@ -152,10 +154,11 @@ async def liquid_probe(
         threshold_pascals: float,
         plunger_impulse_time: float,
         num_baseline_reads: int,
-        output_format: OutputOptions = OutputOptions.can_bus_only,
-        data_files: Optional[Dict[InstrumentProbeType, str]] = None,
         probe: InstrumentProbeType = InstrumentProbeType.PRIMARY,
         force_both_sensors: bool = False,
+        response_queue: Optional[
+            asyncio.Queue[dict[SensorId, list[SensorDataType]]]
```
Comment (sfoster1) [inline api/src/opentrons/hardware_control/backends/flex_protocol.py]: if we're going to use the lower case versions here we should change the rest of the file imo

File: api/src/opentrons/util/logging_config.py [hunk 3]
```diff
@@ -66,6 +75,16 @@ def _host_config(level_value: int) -> Dict[str, Any]:
                 "level": logging.DEBUG,
                 "propagate": False,
             },
+            "opentrons_hardware.sensors.sensor_driver": {
```
Comment (sfoster1) [inline api/src/opentrons/util/logging_config.py]: we should give these sources a special name I think. this makes _all_ logging from that file (given that we typically build loggers with `logging.getLogger(__name__)`) go to this file. if we make the logger mapping keyed by something like `"pipette-pressure-sensor-log"` (obv a constant stored somewhere) then in the sensor driver file we can make one logger with `logging.getLogger(__name__)` for normal kinds of logging and a logger with `logging.getLogger("pipette-pressure-sensor-log")` for logging data. Could even have a separate one for capacitive and pressure sensors, etc

File: api/tests/opentrons/hardware_control/backends/test_ot3_controller.py [hunk 4]
```diff
@@ -725,18 +737,18 @@ async def test_liquid_probe(
...
         )
     except PipetteLiquidNotFoundError:
         # the move raises a liquid not found now since we don't call the move group and it doesn't
         # get any positions back
         pass
     move_groups = mock_move_group_run.call_args_list[0][0][0]._move_groups
-    head_node = axis_to_node(Axis.by_mount(mount))
-    tool_node = sensor_node_for_mount(mount)
     #  in tool_sensors, pipette moves down, then sensor move goes
+    print(move_groups)
```
Comment (sfoster1) [inline api/tests/opentrons/hardware_control/backends/test_ot3_controller.py]: debug print

**Review by sfoster1 — COMMENTED** [review]
Let's make these more special loggers I think

File: api/src/opentrons/util/logging_config.py [hunk 5]
```diff
@@ -106,6 +122,12 @@ def _buildroot_config(level_value: int) -> Dict[str, Any]:
                 "formatter": "message_only",
                 "SYSLOG_IDENTIFIER": "opentrons-api-serial-usbbin",
             },
+            "sensor": {
+                "class": "systemd.journal.JournalHandler",
```
Comment (sfoster1) [inline api/src/opentrons/util/logging_config.py]: IMO we shouldn't send this to the journal - it will be a bunch of data

**Review by sfoster1 — CHANGES_REQUESTED** [review]
Looks good but let's consider whether the data should go in journald or not and add some tests for at least "how many messages am I trying to receive" mechanics and ideally more.

File: api/src/opentrons/hardware_control/backends/flex_protocol.py (hunk 2 above)
Comment (ryanthecoder) [reply api/src/opentrons/hardware_control/backends/flex_protocol.py]: I'm not sure what that means

Comment (sfoster1) [reply api/src/opentrons/hardware_control/backends/flex_protocol.py]: for the types - `dict` instead of `typing.Dict`

**Review by sfoster1 — APPROVED** [review]
Looks good to me!

**Pull request merged**